objectivec/include/ort_training_session.h

```diff
@@ -0,0 +1,242 @@
+ * learning rate to be used throughout the training session.
+ *
+ * @note It does not set the initial learning rate that may be needed by the predefined learning rate schedulers.
+ * To set the initial learning rate for learning rate schedulers, use the `registerLinearLRScheduler` method.
+ *
+ * @param lr The learning rate to be used by the training session.
+ * @param error Optional error information set if an error occurs.
+ * @return YES if the learning rate was set successfully, NO otherwise.
+ */
+- (BOOL)setLearningRate:(float)lr
+                  error:(NSError**)error;
+
+/**
+ * Loads the training session model parameters from a contiguous buffer
+ *
+ * @param buffer Contiguous buffer to load the parameters from.
+ * @param error Optional error information set if an error occurs.
+ * @return YES if the parameters were loaded successfully, NO otherwise.=
+ */
+- (BOOL)fromBufferWithValue:(ORTValue*)buffer
+                      error:(NSError**)error;
+
+/**
+ * Returns a contiguous buffer that holds a copy of all training state parameters.
+ *
+ * @param onlyTrainable If YES, returns a buffer that holds only the trainable parameters, otherwise returns a buffer
+ * that holds all the parameters.
+ * @param error Optional error information set if an error occurs.
+ * @return A contiguous buffer that holds a copy of all training state parameters.
+ */
+- (nullable ORTValue*)toBufferWithTrainable:(BOOL)onlyTrainable
+                                      error:(NSError**)error;
+
+/**
+ * Exports the training session model that can be used for inference.
+ *
+ * If the training session was provided with an eval model, the training session can generate an inference model if it
+ * knows the inference graph outputs. The input inference graph outputs are used to prune the eval model so that the
+ * inference model's outputs align with the provided outputs. The exported model is saved at the path provided and
+ * can be used for inferencing with `ORTSession`.
+ *
+ * @note The method reloads eval model from the path provided to initializer and expects that this path to be valid.
+ *
+ * @param infernceModelPath The path to serialize the inference model.
+ * @param graphOutputNames The names of the outputs that are needed in the inference model.
+ * @param error Optional error information set if an error occurs.
+ * @return YES if the inference model was exported successfully, NO otherwise.
+ */
+- (BOOL)exportModelForInferenceWithOutputPath:(NSString*)infernceModelPath
+                             graphOutputNames:(NSArray<NSString*>*)graphOutputNames
+                                        error:(NSError**)error;
+@end
+
+/**
+ * This function sets the seed for generating random numbers.
+ * Use this function to generate reproducible results. It should be noted that completely reproducible results are not guaranteed.
+ *
+ * @param seed Manual seed to use for random number generation
+ */
+void ORTSetSeed(int64_t seed);
```

Comment:
I think we may need to use `extern "C"` here, or we may hit link issues when this is used from the package.

e.g.:
https://github.com/microsoft/onnxruntime/blob/341484e67c8c8957f02a82d4023ddcdacd890cde/objectivec/include/ort_env.h#L10-L23